internal/providers/vault/issue_257.go

```diff
@@ -0,0 +1,54 @@
+// Copyright 2022 Namespace Labs Inc; All rights reserved.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+
+package vault
+
+import (
+	"bytes"
+	"encoding/json"
+	"io"
+	"net/http"
+	"strconv"
+	"strings"
+
+	"github.com/hashicorp/vault-client-go"
+)
+
+// Temporary workaround until the fix is merged:
+// https://github.com/hashicorp/vault-client-go/pull/260
+func withIssue257Workaround() vault.ClientOption {
+	c := vault.DefaultConfiguration().HTTPClient
+	c.Transport = fix257{rt: c.Transport}
+	return vault.WithHTTPClient(c)
+}
+
+type fix257 struct {
+	rt http.RoundTripper
+}
+
+func (w fix257) RoundTrip(req *http.Request) (*http.Response, error) {
+	res, err := w.rt.RoundTrip(req)
+	if !strings.HasSuffix(req.URL.Path, "/secret-id") ||
```

Comment:
This if looks a bit complex (I would split it to multiple if), but as it is a temporary workaround might be OK.

Reply:
It also has a bug as it doesn't always close res.Body. :-)

Reply:
Coming back to this, it was in fact intentional that the body isn't closed, since we don't consume the bady if we return early (the caller will consume & close it). But it was also confusing… So I added a comment now and moved the close right next to where we consume the body, to make the intention obvious here.